sdk/src/main/java/com/google/cloud/dataflow/sdk/util/common/Counter.java

```diff
@@ -218,7 +220,17 @@
    * Returns the counter's name.
    */
   public String getName() {
-    return name;
+    return this.name.name();
+  }
+
+  /** Returns a Name object which uniquely identifies this counter. */
+  public Name getUniqueName() {
+    return this.name;
+  }
+
+  /** Sets the full name of the counter. */
+  public void setName(Name counterName) {
+    this.name = checkNotNull(counterName);
```

Comment:
Note that this `setName` operation is potentially risky. If the counter is in a map/set (which I think is the case in `CounterSet`) then this could change the `hashCode` / `equals` definition, which would lead to being unable to find the `Counter` in the map/set. Why is this mutation safe?